apps/web/app/.server/service/ethereum-l1/contracts/protocol-upgrade-handler.ts

```diff
@@ -22,10 +23,28 @@ export async function emergencyUpgradeBoardAddress() {
   return upgradeHandler.read.emergencyUpgradeBoard();
 }
 
-export async function getUpgradeStartedEvents(
-  params: Parameters<typeof upgradeHandler.getEvents.UpgradeStarted>[1]
-) {
-  return upgradeHandler.getEvents.UpgradeStarted(undefined, params);
+export async function getUpgradeStartedEvents(params: { fromBlock: bigint; toBlock: bigint }) {
+  const blocksInADay = BigInt(Math.floor((24 * 60 * 60) / EthereumConfig.l1.blockTime));
+
+  const calls = [];
+  let currentFromBlock = params.fromBlock;
+  while (currentFromBlock + blocksInADay < params.toBlock) {
+    calls.push(
+      upgradeHandler.getEvents.UpgradeStarted(undefined, {
+        fromBlock: currentFromBlock,
+        toBlock: currentFromBlock + blocksInADay - 1n,
+      })
+    );
+    currentFromBlock += blocksInADay;
+  }
+  calls.push(
+    upgradeHandler.getEvents.UpgradeStarted(undefined, {
+      fromBlock: currentFromBlock,
+      toBlock: "latest",
```

Comment:
I guess what we really want here is the toBlock that was passed to the function

Reply:
Well, when we call this we always want to get until the latest block. But it's true that this is a little bit cryptic. I can make that change.